src/textinput/styles/textinput.css

```diff
@@ -0,0 +1,40 @@
+@import '../../styles/variables.css';
+
+.root {
+	font: var(--base-font);
+}
+
+.root input {
+	border: 1px solid var(--dojo-shade-gray);
+	padding: var(--input-padding);
+}
+
+.root input:focus {
+	border-color: var(--dojo-light-blue);
+	box-shadow: var(--input-focus-shadow);
+	outline: none;
+}
+
+.root input:disabled {
+	background-color: var(--dojo-off-white);
+	color: var(--dojo-shade-gray);
+}
+
+/* TODO: better required styles */
+.root input:required {
+	border-left-width: 8px;
+}
+
+/* TODO: this needs non-color design as well */
+.invalid input {
+	border-color: var(--dojo-red);
```

Comment:
we should avoid calling out colours directly in our `css` files. better to have `--dojo-error-color` and `--dojo-valid-color` as red and green don't always mean these things.